### DISCUSSION TITLE
Use an own trained SpeechBrain model

### DISCUSSION THREAD

**Discussion by matakanobu** [pyannote/pyannote-audio#920] [Ideas]
Hi, I am trying pyannote2.0 with speechbrain embeddings.

In order to use an own trained speechbrain model, the top directory of SpeechBrain model must be named "speechbrain". 

https://github.com/pyannote/pyannote-audio/blob/bbd5513a1db8bf1e4caec819bfd28ebaf3c398af/pyannote/audio/pipelines/speaker_verification.py#L268-L271

It seems as a bit of a tight limit. For example, how about relaxing it as follows?
```python
    if isinstance(embedding, str) and "speechbrain" in embedding.split("/"):
        return SpeechBrainPretrainedSpeakerEmbedding(embedding, device=device)
    else:
        return PyannoteAudioPretrainedSpeakerEmbedding(embedding, device=device)
```

I think when using the pyannote embeddings model, it is unlikely that the structure of directory contains  "speechbrain".

**Comment by hbredin** [comment]
OK for making things a bit more flexible. I was not happy with the original solution anyway :-)

What about something even looser like `if "speechbrain" in embedding`? 
This is still not perfect but that would probably do the trick...
I'd gladly merge a PR bringing this change.

**Comment by matakanobu** [comment]
> What about something even looser like if "speechbrain" in embedding?

Much better.

> This is still not perfect but that would probably do the trick...

Yes, this is still a workaround...

Sure. I will PR.

**Reply by hbredin** [reply]
I just merged the PR.
